CMakeLists.txt

```diff
@@ -469,68 +469,81 @@ foreach(e ${LIBVNCCLIENT_EXAMPLES})
 endforeach(e ${LIBVNCCLIENT_EXAMPLES})
 
-
 #
 # them tests
 #
-
 if(UNIX)
     set(ADDITIONAL_TEST_LIBS m)
 endif(UNIX)
 
 set(SIMPLETESTS
    cargstest
    copyrecttest
+   wstest
 )
 
-if(CMAKE_USE_PTHREADS_INIT)
-  set(SIMPLETESTS
-      ${SIMPLETESTS}
-      encodingstest
+add_test(NAME cargs COMMAND test_cargstest)
+add_test(NAME websockets_decode COMMAND test_wstest)
+
+if(CMAKE_USE_PTHREADS_INI)
+  list(APPEND SIMPLETESTS encodingstest)
+endif(CMAKE_USE_PTHREADS_INI)
+
+if(FOUND_LIBJPEG_TURBO)
+  list(APPEND SIMPLETESTS tjunittest tjbench)
+  set(tjunittest_add_src 
+      ${TESTS_DIR}/tjutil.c
+      ${TESTS_DIR}/tjutil.h
+      ${COMMON_DIR}/turbojpeg.c
+      ${COMMON_DIR}/turbojpeg.h
      )
-endif(CMAKE_USE_PTHREADS_INIT)
+
+ set(tjbench_add_src
+     ${TESTS_DIR}/tjbench.c
+     ${TESTS_DIR}/tjutil.c
+     ${TESTS_DIR}/tjutil.h
+     ${TESTS_DIR}/bmp.c
+     ${TESTS_DIR}/bmp.h
+     ${COMMON_DIR}/turbojpeg.c
+     ${COMMON_DIR}/turbojpeg.h
+    )
+
+  add_test(NAME turbojpeg COMMAND test_tjunittest)
+endif(FOUND_LIBJPEG_TURBO)
 
 file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/test)
 foreach(t ${SIMPLETESTS})
-  add_executable(test_${t} ${TESTS_DIR}/${t}.c)
+  add_executable(test_${t} ${TESTS_DIR}/${t}.c ${${t}_add_src})
   set_target_properties(test_${t} PROPERTIES OUTPUT_NAME ${t})
   set_target_properties(test_${t} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/test)
   target_link_libraries(test_${t} vncserver vncclient ${ADDITIONAL_TEST_LIBS})
 endforeach(t ${SIMPLETESTS})
 
-if(FOUND_LIBJPEG_TURBO)
-  add_executable(test_tjunittest
-                 ${TESTS_DIR}/tjunittest.c
-                 ${TESTS_DIR}/tjutil.c
-                 ${TESTS_DIR}/tjutil.h
-                 ${COMMON_DIR}/turbojpeg.c
-                 ${COMMON_DIR}/turbojpeg.h
-                )
-  set_target_properties(test_tjunittest PROPERTIES OUTPUT_NAME tjunittest)
-  set_target_properties(test_tjunittest PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/test)
-  target_link_libraries(test_tjunittest vncserver vncclient ${ADDITIONAL_TEST_LIBS})
-
-  add_executable(test_tjbench
-                 ${TESTS_DIR}/tjbench.c
-                 ${TESTS_DIR}/tjutil.c
-                 ${TESTS_DIR}/tjutil.h
-                 ${TESTS_DIR}/bmp.c
-                 ${TESTS_DIR}/bmp.h
-                 ${COMMON_DIR}/turbojpeg.c
-                 ${COMMON_DIR}/turbojpeg.h
-                )
-  set_target_properties(test_tjbench PROPERTIES OUTPUT_NAME tjbench)
-  set_target_properties(test_tjbench PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/test)
-  target_link_libraries(test_tjbench vncserver vncclient ${ADDITIONAL_TEST_LIBS})
-
-endif(FOUND_LIBJPEG_TURBO)
-
-add_test(NAME cargs COMMAND test_cargstest)
-if(FOUND_LIBJPEG_TURBO)
-    add_test(NAME turbojpeg COMMAND test_tjunittest)
-endif(FOUND_LIBJPEG_TURBO)
-
-
+#if(FOUND_LIBJPEG_TURBO)
```

Comment:
With "untouched" you mean, recreating the original (imho not-so-nice) version? or the one I from my commit?